skaha/src/main/java/org/opencadc/skaha/K8SUtil.java

```diff
@@ -203,4 +211,42 @@ public static String getRedisPort() {
     public static String getUserHome() {
         return System.getProperty("user.home");
     }
+
+    public static String getVNCURL(String host, String sessionID) {
+        final String cleanSessionID = Objects.requireNonNull(sessionID);
+        return String.format(
+                K8SUtil.VNC_URL_TEMPLATE, Objects.requireNonNull(host), cleanSessionID, cleanSessionID, cleanSessionID);
+    }
+
+    public static String getCartaURL(String host, String sessionID, boolean altSocketUrl) {
+        return altSocketUrl ? K8SUtil.getCartaAltSocketURL(host, sessionID) : K8SUtil.getCartaURL(host, sessionID);
+    }
+
+    private static String getCartaURL(final String host, final String sessionID) {
+        return String.format(
+                K8SUtil.CARTA_URL_TEMPLATE, Objects.requireNonNull(host), Objects.requireNonNull(sessionID));
+    }
+
+    private static String getCartaAltSocketURL(final String host, final String sessionID) {
+        final String cleanSessionID = Objects.requireNonNull(sessionID);
+        final String cleanHost = Objects.requireNonNull(host);
+        return String.format(
+                K8SUtil.CARTA_ALT_SOCKET_URL_TEMPLATE, cleanHost, cleanSessionID, cleanHost, cleanSessionID);
+    }
+
+    public static String getNotebookURL(String host, String sessionID, String userid, String skahaTLD) {
+        final String cleanSessionID = Objects.requireNonNull(sessionID);
+        return String.format(
```

Comment:
Rather than using `String.format` we should use a library, to catch malformed URLS and also cases when sometimes you have `//path//path/to/something` etc.